File: Friends and Candies/Friends_and_Candies.py
def Friends_and_Candies():
    total_friends = int(input())
    candies_friend_have= list(map(int,input().split()))
    total_candies = sum(candies_friend_have)
    if total_candies % total_friends != 0:
        return -1 
    to_each_friend = total_candies // total_friends
   
    minimum_select = 0
    
    
    for i in range(total_friends):
        if candies_friend_have[i] > to_each_friend:
            minimum_select += 1 
    return minimum_select
        
remained_test_cases = 1 
remained_test_cases = int(input())
while remained_test_cases > 0:
    print(Friends_and_Candies())
    remained_test_cases -= 1 

File: Friends and Candies/Friends_and_Candies.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;

int Friends_and_Candies()
{
   ll total_friends;
   cin>>total_friends;
   ll candies_friend_have[total_friends];
   ll total_candies = 0;
   
   for(ll i = 0 ;i < total_friends; i++)
   {  cin>>candies_friend_have[i];
      total_candies += candies_friend_have[i];
       
   }
   
   if(total_candies % total_friends != 0)
   return -1;
   ll each_get = total_candies / total_friends;
   ll  minimum_select = 0;
   
   for(ll i = 0; i < total_friends ;i++ )
   {
       if(candies_friend_have[i] > each_get)
       minimum_select += 1 ;
   }
   
   return minimum_select;
   
   
}


int main() {
	int remained_test_cases = 1 ;
	cin>>remained_test_cases;
	while(remained_test_cases > 0)
	{   cout<<Friends_and_Candies()<<"\n";
	    remained_test_cases--;
	}
	return 0;
}
